fix: do not set placeholder to empty string by default

web-padawan opened the pull request:
## Description

Fixes #7551

Changed to align with `InputControlMixin` by removing default empty string value, so that we don't have `placeholder` attribute set by default. Also fixed the logic for restoring placeholder property in selected items observer.

Note, the property has `reflectToAttribute` so technically setting empty string can still cause that to happen (which is the case for all input fields). I don't think we can easily fix that without a breaking change.

## Type of change

- Bugfix

vursen commented:
The change appears to introduce a regression: after deselecting all items, the original placeholder doesn't return, and instead, the last deselected item becomes shown in the placeholder.


https://github.com/user-attachments/assets/0114c3e1-c706-4503-8293-4d728593f4e2


```html
<vaadin-multi-select-combo-box placeholder="Placeholder"></vaadin-multi-select-combo-box>

<script>
  const comboBox = document.querySelector('vaadin-multi-select-combo-box');
  comboBox.items = ['Item 1'];
  comboBox.selectedItems = ['Item 1'];
</script>
```

web-padawan commented:
> Looks like the change results in a regression: after deselecting all items, the original placeholder doesn't come back

Thanks, this actually is related to the fact that in your example (and the dev page) we use `<script>` without `type` set to `module` (and therefore both `placeholder` and `selectedItems` are set before the component gets upgraded.

Added a fix for this case and updated tests to reproduce it (by replacing `fixtureSync()` usage with manual adding to DOM).

vursen commented:
> this actually is related to the fact that in your example (and the dev page) we use <script> without type set to module (and therefore both placeholder and selectedItems are set before the component gets upgraded.

Looks like this didn't entirely fix the regression: now, when deselecting all items, the placeholder doesn't clear and shows the last deselected item again.

```html
<vaadin-multi-select-combo-box></vaadin-multi-select-combo-box>

<script>
  const comboBox = document.querySelector('vaadin-multi-select-combo-box');
  comboBox.items = ['Item 1'];
  comboBox.selectedItems = ['Item 1'];
</script>```

web-padawan commented:
> When deselecting all items without an initial placeholder, the placeholder doesn't clear 

Thanks, fixed and added a test with setting `placeholder` to `undefined`.
